bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/cloud/MiCloudConnector.java

```diff
@@ -154,22 +155,26 @@ public String getMapUrl(String vacuumMap, String country) throws MiCloudExceptio
         String mapResponse = request(url, map);
         logger.trace("response: {}", mapResponse);
         String errorMsg = "";
-        JsonElement response = PARSER.parse(mapResponse);
-        if (response.isJsonObject()) {
-            logger.debug("Received  JSON message {}", response.toString());
-            if (response.getAsJsonObject().has("result") && response.getAsJsonObject().get("result").isJsonObject()) {
-                JsonObject jo = response.getAsJsonObject().get("result").getAsJsonObject();
-                if (jo.has("url")) {
-                    String mapUrl = jo.get("url").getAsString();
-                    return mapUrl != null ? mapUrl : "";
+        try {
+            JsonElement response = PARSER.parse(mapResponse);
+            if (response.isJsonObject()) {
+                logger.debug("Received  JSON message {}", response.toString());
```

Comment:
You don't have to call `toString` it will be called automatically, but only when the logging-level actually applies.

Reply:
agree.done